qemu/tests/slof_hugepage.py

```diff
@@ -0,0 +1,136 @@
+
+    hugepage_size = huge_page.get_hugepage_size()
+    logging.debug('Hugepage size is %skB in host.' % hugepage_size)
+
+    huge_page.target_hugepages = int((int(size) * 1024) // hugepage_size)
+    logging.debug('Set hugepages to %d pages in host.'
+                  % huge_page.target_hugepages)
+    huge_page.set_node_num_huge_pages(huge_page.target_hugepages,
+                                      0, hugepage_size)
+
+    error_context.context('mount hugepages to %s'
+                          % huge_page.hugepage_path, logging.info)
+    huge_page.mount_hugepage_fs()
+
+
+def _check_mem_increase(session, params, orig_mem):
+    """Check the size of memory increased in timeout."""
+    end_time = time.time() + float(params.get('plug_timeout', 5))
+    while time.time() < end_time:
+        increase_mem = int(
+            utils_numeric.normalize_data_size(params['size_plug'], 'B'))
+        new_mem = int(session.cmd_output(cmd=params['free_mem_cmd']))
+        if (new_mem - orig_mem) == increase_mem:
+            error_context.context(
+                'Get guest free memory size after hotplug pc-dimm.',
+                logging.info)
+            logging.debug('Guest free memory size is %d bytes' % new_mem)
+            logging.info(
+                "Guest memory size is increased %s." % params['size_plug'])
+            return True
+        time.sleep(1)
+    return False
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Verify SLOF info with hugepage.
+
+    Step:
+     1. Assign definite size hugepage and mount it in host.
+     2. Boot a guest by following ways:
+         a. hugepage as backing file
+         b. hugepage not as backing file
+        then Check if any error info in output of SLOF.
+     3. Get the size of memory inside guest.
+     4. Hot plug pc-dimm by QMP.
+     5. Get the size of memory after hot plug pc-dimm inside guest,
+        then check the different value of memory.
+     6. Reboot guest.
+     7. Guest could login successfully.
+     8. Guest could ping external host ip.
+
+    :param test: Qemu test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+    def _wait_for_login(cur_pos=0):
+        """Wait for login guest."""
+        content, next_pos = slof.wait_for_loaded(vm, test, cur_pos)
+        error_context.context("Check the output of SLOF.", logging.info)
+        slof.check_error(test, content)
+
+        error_context.context("Try to log into guest '%s'." % vm.name,
+                              logging.info)
+        timeout = float(params.get("login_timeout", 240))
+        session = vm.wait_for_login(timeout=timeout)
+        logging.info("log into guest '%s' successfully." % vm.name)
+        return session, next_pos
+
+    _setup_hugepage(params)
+
+    params['start_vm'] = 'yes'
+    env_process.preprocess_vm(test, params, env, params["main_vm"])
```

Comment:
Do you need to add `not_preprocess = yes`? so it dose not `preprocess_vm` twice.

Reply:
Thanks,  have updated.